cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceTaskContextProvider.java

```diff
@@ -55,7 +55,7 @@
   private BasicMapReduceTaskContext context;
   private AbstractMapReduceTaskContextBuilder contextBuilder;
 
-  public MapReduceTaskContextProvider(TaskAttemptContext context, MapReduceMetrics.TaskType type) {
+  public MapReduceTaskContextProvider(TaskAttemptContext context, @Nullable MapReduceMetrics.TaskType type) {
```

Comment:
In the FileOutputFormat#commitJob, its not from a mapper nor a reducer, but we still instantiate the MapReduceTaskContext, to get access to dataset.